VERSION "1.0"

NS_ :
    NS_DESC_
    CM_
    BA_DEF_
    BA_
    VAL_
    CAT_DEF_
    CAT_
    FILTER
    BA_DEF_DEF_
    EV_DATA_
    ENVVAR_DATA_
    SGTYPE_
    SGTYPE_VAL_
    BA_DEF_SGTYPE_
    BA_SGTYPE_
    SIG_TYPE_REF_
    VAL_TABLE_
    SIG_GROUP_
    SIG_VALTYPE_
    SIGTYPE_VALTYPE_
    BO_TX_BU_
    BA_DEF_REL_
    BA_REL_
    BA_DEF_DEF_REL_
    BU_SG_REL_
    BU_EV_REL_
    BU_BO_REL_
    SG_MUL_VAL_

BS_: 500000,1,10

BU_: ECU1 ECU2 ECU3

BO_ 100 EngineData: 8 ECU1
 SG_ EngineSpeed : 0|16@1+ (0.1,0) [0|6500] "rpm" ECU2,ECU3
 SG_ EngineTemp : 16|8@1+ (1,-40) [-40|215] "degC" ECU2
 SG_ EngineLoad : 24|8@1+ (1,0) [0|100] "%" ECU2,ECU3

BO_ 200 TransmissionData: 6 ECU2
 SG_ GearPosition : 0|4@1+ (1,0) [0|8] "" ECU1,ECU3
 SG_ TransmissionMode M : 4|2@1+ (1,0) [0|3] "" ECU1,ECU3
 SG_ TransmissionTemp : 8|8@1+ (1,-40) [-40|215] "degC" ECU1
 SG_ TransmissionSpeed : 16|16@1+ (0.1,0) [0|6500] "rpm" ECU1,ECU3
 SG_ TransmissionInfo m0 : 32|8@1+ (1,0) [0|255] "" ECU1
 SG_ TransmissionPressure m1 : 32|8@1+ (1,0) [0|255] "kPa" ECU1

CM_ BU_ ECU1 "Engine Control Unit";
CM_ BU_ ECU2 "Transmission Control Unit";
CM_ BU_ ECU3 "Diagnostic Unit";
CM_ BO_ 100 "Engine data message";
CM_ SG_ 100 EngineSpeed "Engine speed in RPM";
CM_ SG_ 100 EngineTemp "Engine temperature in degrees Celsius";
CM_ SG_ 100 EngineLoad "Engine load as percentage";
CM_ BO_ 200 "Transmission data message";
CM_ SG_ 200 GearPosition "Current gear position";

BA_DEF_ SG_ "SignalType" STRING ;
BA_DEF_ BO_ "GenMsgCycleTime" INT 0 10000;
BA_DEF_DEF_ "SignalType" "";
BA_DEF_DEF_ "GenMsgCycleTime" 100;
BA_ "GenMsgCycleTime" BO_ 100 100;
BA_ "GenMsgCycleTime" BO_ 200 200;

VAL_ 200 GearPosition 0 "Neutral" 1 "First" 2 "Second" 3 "Third" 4 "Fourth" 5 "Fifth" 6 "Sixth" 7 "Reverse" 8 "Park";
VAL_ 200 TransmissionMode 0 "Normal" 1 "Sport" 2 "Economy" 3 "Winter"; 